### ISSUE TITLE
[Feature Request] alarm profile logic to trigger if data key is not updated for XXX minutes

### ISSUE THREAD

**Issue by t0mZ4** [thingsboard/thingsboard#8614]
some devices are updated by multiple sources, or some devices might go into error state and only report keepAlive but not the important data key.

so i would like to set an alarm, if ___ not updated for ___ minutes

for example, if timeseries Temperature not updated for 60 minutes

trigger alert

hope you can add it

@ashvayka

**Comment by ashvayka** [comment]
Yes, we can! But I can't promise the release date. I will plan to add it to 3.6

**Comment by lfjm2022** [comment]
I am also interested in this feature. What is the priority level for it's development? @ashvayka

**Comment by ashvayka** [comment]
We are now redesigning the Alarm Rules and moving them from Device Profile to a separate service. We plan to add this feature in the initial release of the new Alarm Rule Service which is targeted for 3.7. The Alarm Rule Service is a main feature of 3.7 release so I am sure it will arrive

**Comment by t0mZ4** [comment]
That's great, looking forward to getting this feature and to see the new
Alarm Logic Engine!

<https://iot-experts.net/>

Tom Zamir

Owner

***@***.***

+972-55-660-1239 <+972556601239>

Pardes Hanna, Israel

iot-experts.net

<https://www.linkedin.com/r/tomzamir>

**Comment by GregAscolab** [comment]
> We are now redesigning the Alarm Rules and moving them from Device Profile to a separate service. We plan to add this feature in the initial release of the new Alarm Rule Service which is targeted for 3.7. The Alarm Rule Service is a main feature of 3.7 release so I am sure it will arrive

**Any news on this new "Alarm service" ?** (nothing 3.7 yet...)
Have got lot of "alarms to generate" and due to missing "OR logic" and other limitation in "alarms rules", I need to create very complicated "rules engines"...

Other discussions :
https://github.com/thingsboard/thingsboard/issues/8681#issuecomment-1569907130
https://github.com/thingsboard/thingsboard/issues/7227#issuecomment-1326581740

**Comment by GregAscolab** [comment]
Any news on this feature roadmap ?

**Comment by ViacheslavKlimov** [comment]
This will be a part of **4.3** release which is planned for mid-January.

**Comment by mykhailokornieiev** [comment]
Hi @t0mZ4 @GregAscolab 
I can confirm that this feature is present in the upcoming v4.3 as mentioned by @ViacheslavKlimov 
It will be available as a new alarm filter operator "missing for" with a specified time value:

<img width="892" height="531" alt="Image" src="https://github.com/user-attachments/assets/470c4d86-a5d4-4968-ba6d-f78962f06826" />

Since the feature is already implemented in the upcoming release, I close this ticket as solved. 
Thank you for the suggestion and have a nice day.
